tests/msc2716_test.go

```diff
@@ -255,19 +255,87 @@ func TestImportHistoricalMessages(t *testing.T) {
 			})
 		})
 
+		t.Run("Backfill still works after many batches are imported", func(t *testing.T) {
+			t.Parallel()
+
+			roomID := as.CreateRoom(t, createPublicRoomOpts)
+			alice.JoinRoom(t, roomID, nil)
+
+			// Create some normal messages in the timeline
+			eventIDsBefore := createMessagesInRoom(t, alice, roomID, 2, "eventIDsBefore")
+			eventIdBefore := eventIDsBefore[len(eventIDsBefore)-1]
+			timeAfterEventBefore := time.Now()
+
+			// We chose the magic number 11 because Synapse currently limits the
+			// backfill extremities to 5. 10 also seemed like a round number someone
+			// could pick for other homeserver implementations so I just did 10+1 to
+			// make sure it also worked in that case.
+			//numBatches := 11
+			numBatches := 2
```

Comment:
Lovely comment but then you seem to use 2. Why?

Reply:
See:

> I used `numBatches=2` during testing because it's much faster to see results while developing.
>
> https://github.com/matrix-org/complement/pull/214#discussion_r746829737

Running the tests against Synapse is extremely slow atm, https://github.com/matrix-org/complement/pull/214#discussion_r741678240